pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1149,7 +1149,7 @@ func applyUpstreamTLSSettings(opts *buildClusterOpts, tls *networking.TLSSetting
 		}
 
 		// Fallback to file mount secret instead of SDS if meshConfig.sdsUdsPath isn't set or tls.mode is TLSSettings_MUTUAL.
-		if opts.push.Mesh.SdsUdsPath == "" || tls.Mode == networking.TLSSettings_MUTUAL {
+		if !node.Metadata.SdsEnabled || opts.push.Mesh.SdsUdsPath == "" || tls.Mode == networking.TLSSettings_MUTUAL {
```

Comment:
Looks like `SdsUdsPath` is still being used from mesh config - Should that also be taken from metadata?

Reply:
We don't have it in the metadata, so we don't really have a choice here since we cannot go back in time to old versions and add it - this is primarily about supporting upgrade from 1.4 -> 1.5. Should it have been done originally? probably. But I think now we have deprecated the mesh config field and will just use a hardcoded value

Reply:
I see. Makes sense.